openmmapi/include/openmm/NoseHooverIntegrator.h

```diff
@@ -0,0 +1,279 @@
+    /**
+     * Get the temperature of the i-th chain for controling absolute particle motion (in Kelvin).
+     * 
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     * 
+     * @return the temperature.
+     */
+    double getTemperature(int chainID=0) const;
+    /**
+     * set the (absolute motion) temperature of the i-th chain.
+     *
+     * @param temperature the temperature for the Nose-Hoover chain thermostat (in Kelvin).
+     * @param chainID The id of the Nose-Hoover chain thermostat for which the temperature is set (default=0).
+     */
+    void setTemperature(double temperature, int chainID=0);
+    /**
+     * Get the temperature of the i-th chain for controling pairs' relative particle motion (in Kelvin).
+     *
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     *
+     * @return the temperature.
+     */
+    double getRelativeTemperature(int chainID=0) const;
+    /**
+     * set the (relative pair motion) temperature of the i-th chain.
+     *
+     * @param temperature the temperature for the Nose-Hoover chain thermostat (in Kelvin).
+     * @param chainID The id of the Nose-Hoover chain thermostat for which the temperature is set (default=0).
+     */
+    void setRelativeTemperature(double temperature, int chainID=0);
+    /**
+     * Get the collision frequency for absolute motion of the i-th chain (in 1/picosecond).
+     * 
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     *
+     * @return the collision frequency.
+     */
+    double getCollisionFrequency(int chainID=0) const;
+    /**
+     * Set the collision frequency for absolute motion of the i-th chain.
+     *
+     * @param frequency the collision frequency in picosecond.
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     */
+    void setCollisionFrequency(double frequency, int chainID=0);
+    /**
+     * Get the collision frequency for pairs' relative motion of the i-th chain (in 1/picosecond).
+     *
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     *
+     * @return the collision frequency.
+     */
+    double getRelativeCollisionFrequency(int chainID=0) const;
+    /**
+     * Set the collision frequency for pairs' relative motion of the i-th chain.
+     *
+     * @param frequency the collision frequency in picosecond.
+     * @param chainID the index of the Nose-Hoover chain thermostat (default=0).
+     */
+    void setRelativeCollisionFrequency(double frequency, int chainID=0);
+    /**
+     * Compute the total (potential + kinetic) heat bath energy for all heat baths
+     * associated with this integrator, at the current time.
+     */
+    double computeHeatBathEnergy();
+    /**
+     * Get the number of Nose-Hoover chains registered with this integrator.
+     */
+    int getNumNoseHooverThermostats() const {
```

Comment:
The method names are slightly inconsistent: `addThermostat()` but `getNumNoseHooverThermostats()`.  How about making all of them consistent: `addThermostat()`, `getThermostat()`, `getNumThermostats()`.

Reply:
OK